staging/src/k8s.io/client-go/tools/cache/listwatch.go

```diff
@@ -93,95 +102,96 @@ func NewFilteredListWatchFromClient(c Getter, resource string, namespace string,
 	return &ListWatch{ListFunc: listFunc, WatchFunc: watchFunc}
 }
 
-func timeoutFromListOptions(options metav1.ListOptions) time.Duration {
-	if options.TimeoutSeconds != nil {
-		return time.Duration(*options.TimeoutSeconds) * time.Second
-	}
-	return 0
+// applyUserListOptions applies only those options that are allowed to be set in List and Watch for an Informer
+func applyUserListOptions(options metav1.ListOptions, to *metav1.ListOptions) {
+	/*
+	 * Can't copy:
+	 * - ResourceVersion
+	 * - Timeout
+	 * as these are used by informers internally
+	 */
+
+	to.LabelSelector = options.LabelSelector
+	to.FieldSelector = options.FieldSelector
+	to.IncludeUninitialized = options.IncludeUninitialized
 }
 
-// List a set of apiserver resources
-func (lw *ListWatch) List(options metav1.ListOptions) (runtime.Object, error) {
-	if !lw.DisableChunking {
-		return pager.New(pager.SimplePageFunc(lw.ListFunc)).List(context.TODO(), options)
-	}
-	return lw.ListFunc(options)
-}
-
-// Watch a set of apiserver resources
-func (lw *ListWatch) Watch(options metav1.ListOptions) (watch.Interface, error) {
-	return lw.WatchFunc(options)
-}
+// NewListWatchFromMethods creates a ListWatch object from a client supporting List and Watch methods.
+// Because Golang doesn't have templates/generics to allow us to accept an object implementing List+Watch interface but differing
+// in return types the client's type is erased to in interface{} and checked in runtime using reflection.
+func NewListWatchFromMethods(listerWatcher interface{}, userListOptions metav1.ListOptions) *ListWatch {
```

Comment:
I would really like to see listerWatcher being of type ListerWatcher. What is the exact problem that you have with that?

Reply:
I think the issue was that ListerWatcher return types don't match as List and Watch method return types for every object differ there from runtime.Object, so you can't put the method there directly. Anyways I am dropping it.